plugin/pkg/admission/serviceplan/changevalidator/admission.go

```diff
@@ -72,7 +72,11 @@ func (d *denyPlanChangeIfNotUpdatable) Admit(a admission.Attributes) error {
 		return apierrors.NewBadRequest("Resource was marked with kind Instance but was unable to be converted")
 	}
 
-	sc, err := d.scLister.Get(instance.Spec.ClusterServiceClassExternalName)
+	if instance.Spec.ClusterServiceClassRef == nil {
+		return nil // user chose a service class that doesn't exist
+	}
+
+	sc, err := d.scLister.Get(instance.Spec.ClusterServiceClassRef.Name)
 	if err != nil {
```

Comment:
This LGTM, except that it doesn't deal with the case where the user has specified the kubernetes names instead of the external names, which I think should be pretty easy to add.  Add coverage for that and this LGTM

Reply:
Actually, the additive change I described above can be a follow-up.  It's more important to get the fix here in ASAP to me.

Reply:
+1 to @pmorie 's follow-up recommendation.

Reply:
I need a bit of clarification here. It appears that as long as the cluster service class reference is set, that if the external name isn't available the k8s name will be used instead. However, I'm not sure what guarantees are offered by the informers HasSynced method, specifically with regard to the controller updating the class reference. (I tried porting the controller look up code to the admission controller, but realized I couldn't fully do it due to the List interface only accepting a label selector and not ListOptions.)

Reply:
I talked to Paul and basically if the reference hasn't been updated by the controller, there's not much to do. So I don't think a follow up is necessary after all (at least specifically for looking up the k8s name, as that is already handled).